cegs_portal/search/templates/search/v1/dna_feature.html

```diff
@@ -506,25 +525,62 @@ <h1>
         {% endfor %}
     ];
 
-    g("feature-dist-submit").onclick = function() {
-        let dist = Number(g("feature-dist").value);
-        let fetches = assemblies.map(a => {
-            let search_start = Math.max(a.start - dist, 0);
-            let search_end = a.end + dist;
-            return fetch(`/search/featureloc/${a.chromo}/${search_start}/${search_end}?assembly=${a.assembly}&search_type=overlap&accept=application/json`)
-                .then(response => {
-                    return response.json()
-                });
-        })
-        Promise.all(fetches)
-            .then(data => {
-                let features = [];
-                for (const d of data) {
-                    features.push(...d);
-                }
-                g("dnafeature").replaceWith(featureTable(features, "No nearby DNA Features"))
+    function fetchDataAndRenderTable() {
+    let dist = Number(g("feature-dist").value);
+    let fetches = assemblies.map(a => {
+        let search_start = Math.max(a.start - dist, 0);
+        let search_end = a.end + dist;
+        let url = `/search/featureloc/${a.chromo}/${search_start}/${search_end}?assembly=${a.assembly}&search_type=overlap&accept=application/json`;
+
+        let checkboxes = document.querySelectorAll('input[name="feature_type"]:checked');
+        checkboxes.forEach(checkbox => {
+            url += `&feature_type=${checkbox.id}`;
+        });
+
+        return fetch(url)
+            .then(response => {
+                return response.json()
             });
+    })
+    Promise.all(fetches)
+        .then(data => {
+            let features = [];
+            for (const d of data) {
+                features.push(...d);
+            }
+            let table = g("dnafeature");
+            while (table.firstChild) {
+                table.removeChild(table.firstChild);
+            }
+
+    if (features.length === 0) {
+
+    let checkboxes = document.querySelectorAll('input[name="feature_type"]:checked');
+    checkboxes.forEach(checkbox => {
+        checkbox.disabled = true;
+    });
+
+    let parent = document.getElementById('tabs-find-features');
+    parent.style = "display: grid; justify-items: center;";
+
+    let message = document.createElement('div');
+    message.className = "table-title";
+    message.style = "display: flex; align-items: center; justify-content: center";
+    message.textContent = "DNA Feature Type Not Found";
+
+    let container = document.createElement('div');
+    container.className = "container min-w-full";
+    container.style = "margin-left: auto; margin-right: auto; width: 50%;";
+    container.appendChild(message);
+
+    table.appendChild(container);
```

Comment:
You can use the `e` shortcut here:

```suggestion
    table.appendChild(
    e(
        "div",
        {class: "container min-w-full", style: "margin-left: auto; margin-right: auto; width: 50%;"},
        e(
            "div",
            {class: "table-title", style: "display: flex; align-items: center; justify-content: center"},
            "DNA Feature Type Not Found"
        )
    )
);
```